aws/middleware/middleware.go

```diff
@@ -166,3 +180,45 @@ func AddRawResponseToMetadata(stack *middleware.Stack) error {
 func GetRawResponse(metadata middleware.Metadata) interface{} {
 	return metadata.Get(rawResponseKey{})
 }
+
+// AddTimeOffsetBuildMiddleware sets a value representing clock skew on the request context.
+// This can be read by other operations (such as signing) to correct the date value they send
+// on the request
+type AddTimeOffsetBuildMiddleware struct {
```

Comment:
We almost certainly want this under the blanket `internal/` as well. In general, we just don't export things unless we know we absolutely need to.

Reply:
Moved to internal